src/Resources/contao/controllers/BackendIndex.php

```diff
@@ -32,35 +45,12 @@ class BackendIndex extends \Backend
 	 */
 	public function __construct()
 	{
+		$this->container = System::getContainer();
+		$this->flashBag = $this->container->get('session')->getFlashBag();
+
 		$this->import('BackendUser', 'User');
 		parent::__construct();
 
-		// Login
-		if ($this->User->login())
-		{
-			$strUrl = 'contao/main.php';
-
-			// Redirect to the last page visited
-			if (\Input::get('referer', true) != '')
-			{
-				$strUrl = base64_decode(\Input::get('referer', true));
```

Comment:
@bytehead can you check my old implementation. Pretty sure I fixed it there, but I can't remember how exactly.